src/algebra/roption_instances.lean

```diff
@@ -0,0 +1,99 @@
+/-
+Copyright (c) 2019 Hoang Le Truong. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Hoang Le Truong.
+
+ roption α  can see as  Prop × α. (Prop, ∧) has natural algebra structure. Therefore if α is a semigroup, a comm_semigroup, 
+ a comm_monoid,a add_semigroup, a add_comm_semigroup, a add_comm_monoid, a mul_action, a distrib_mul_action, so is roption α. 
+-/
+
+import   algebra.module data.pfun
+
+universes u v w
+variables {α : Type u} {β : Type v} {γ : Type w} 
+
+namespace roption
+noncomputable theory
+
+instance [has_zero α] : has_zero (roption α) := ⟨some (0:α)⟩
+@[simp] lemma zero_def [has_zero α] : (0 : roption α ) = ⟨true, λ _, (0:α)⟩ := rfl
```

Comment:
These lemmas are fine. But they certainly shouldn't be tagged with `@[simp]`. Otherwise `simp` will give horrible results.

Reply:
Done